packages/nextjs/src/config/types.ts

```diff
@@ -17,6 +17,11 @@ export type NextConfigObject = {
   [key: string]: unknown;
 };
 
+export type NextConfigFunction = (
+  phase: string,
```

Comment:
this could be an interesting thing to experiment with in the future, like for ex. we construct different webpack configs based on dev/prod/export (if I'm understanding this correctly). https://github.com/vercel/next.js/blob/canary/packages/next/next-server/lib/constants.ts#L1-L4

Reply:
You are. I think we might wait if/until it proves a problem, though, as I don't believe those things influence the parts of the config we muck with.